### PR TITLE
Migrate to Java 21

### PR DESCRIPTION
**Pull request by ravinperera00**
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/websubhub/ModuleUtils.java [hunk 1]
```diff
@@ -37,4 +42,39 @@ public static void setModule(Environment environment) {
...
     }
+
+    public static void notifySuccess(CompletableFuture<Object> future, Object result) {
+        if (result instanceof BError) {
+            BError error = (BError) result;
+            if (!isModuleDefinedError(error)) {
+                error.printStackTrace();
+            }
+        }
+
```
Comment (warunalakshitha) [inline native/src/main/java/io/ballerina/stdlib/websubhub/ModuleUtils.java:53]: Lets remove this new line

File: native/src/main/java/io/ballerina/stdlib/websubhub/NativeHttpToWebsubhubAdaptor.java [hunk 2]
```diff
@@ -200,20 +200,27 @@ private static boolean isReadOnlyParam(BObject serviceObj, String remoteMethod)
...
+            CompletableFuture<Object> balFuture = new CompletableFuture<>();
+            Module module = ModuleUtils.getModule();
+            StrandMetadata metadata = new StrandMetadata(module.getOrg(), module.getName(), module.getVersion(),
+                    parentFunctionName);
+            ObjectType serviceType = (ObjectType) TypeUtils.getReferredType(TypeUtils.getType(bHubService));
+            Object result;
+            try {
+                if (serviceType.isIsolated() && serviceType.isIsolated(remoteFunctionName)) {
+                    result = env.getRuntime().startIsolatedWorker(
+                            bHubService, remoteFunctionName, null, metadata, null, args).get();
```
[the remark this answers is not in the document]
Comment (warunalakshitha) [reply native/src/main/java/io/ballerina/stdlib/websubhub/NativeHttpToWebsubhubAdaptor.java:213]: shall we break from args not (. We can do same for other lines too

File: .github/workflows/fossa_scan.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: fossas/fossa-action@main
+      - uses: fossas/fossa-action@java21
```
Comment (warunalakshitha) [inline .github/workflows/fossa_scan.yml:11]: We should only change template.yml@main to template.yml@java21

File: native/src/main/java/io/ballerina/stdlib/websubhub/ModuleUtils.java (hunk 1 above)
Comment (ravinperera00) [reply native/src/main/java/io/ballerina/stdlib/websubhub/ModuleUtils.java]: Fixed with 83c65f98306e0da9824443382f65f035661ea530

File: native/src/main/java/io/ballerina/stdlib/websubhub/NativeHttpToWebsubhubAdaptor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ravinperera00) [reply native/src/main/java/io/ballerina/stdlib/websubhub/NativeHttpToWebsubhubAdaptor.java]: Fixed with 83c65f98306e0da9824443382f65f035661ea530

File: .github/workflows/fossa_scan.yml (hunk 3 above)
Comment (ravinperera00) [reply .github/workflows/fossa_scan.yml]: Fixed with 83c65f98306e0da9824443382f65f035661ea530

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**